sdk/samples/iot/paho_iot_hub_pnp_component_sample.c

```diff
@@ -595,77 +658,142 @@ static void disconnect_mqtt_client_from_iot_hub(void)
   MQTTClient_destroy(&mqtt_client);
 }
 
-static void mqtt_publish_message(const char* topic, az_span payload, int qos)
+static void set_publish_topic(
```

Comment:
I'd still rather have these `_get_publish_topic` calls in whatever function is calling this to show the explicit API calls.

Reply:
Then it seems we should also discard the publish_mqtt_message wrapper function which abstracts the explicit API call to MQTT.  Unless you are differentiating between MQTT and IoT Hub?  However, the call to pnp_get_telemetry_publish_topic is also an abstraction of the API.  So I'm not clear why we abstract certain IoT Hub calls and not others.